clouddriver-sql/src/main/kotlin/com/netflix/spinnaker/clouddriver/sql/SqlTaskRepository.kt

```diff
@@ -0,0 +1,252 @@
+/*
+ * Copyright 2018 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.netflix.spinnaker.clouddriver.sql
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.netflix.spinnaker.clouddriver.core.ClouddriverHostname
+import com.netflix.spinnaker.clouddriver.data.task.DefaultTaskStatus
+import com.netflix.spinnaker.clouddriver.data.task.Status
+import com.netflix.spinnaker.clouddriver.data.task.Task
+import com.netflix.spinnaker.clouddriver.data.task.TaskRepository
+import com.netflix.spinnaker.clouddriver.data.task.TaskState
+import com.netflix.spinnaker.clouddriver.data.task.TaskState.FAILED
+import com.netflix.spinnaker.clouddriver.data.task.TaskState.STARTED
+import com.netflix.spinnaker.kork.sql.config.SqlRetryProperties
+import org.jooq.DSLContext
+import org.jooq.Record
+import org.jooq.Select
+import org.jooq.impl.DSL
+import org.jooq.impl.DSL.field
+import org.jooq.impl.DSL.sql
+import org.slf4j.LoggerFactory
+import java.time.Clock
+import java.util.UUID
+
+class SqlTaskRepository(
+  private val jooq: DSLContext,
+  private val mapper: ObjectMapper,
+  private val clock: Clock,
+  private val sqlRetryProperties: SqlRetryProperties
+) : TaskRepository {
+
+  private val log = LoggerFactory.getLogger(javaClass)
+
+  init {
+    log.info("Using ${javaClass.simpleName}")
+  }
+
+  override fun create(phase: String, status: String): Task {
+    return create(phase, status, UUID.randomUUID().toString())
+  }
+
+  override fun create(phase: String, status: String, clientRequestId: String): Task {
+    var task = SqlTask(UUID.randomUUID().toString(), ClouddriverHostname.ID, clientRequestId, clock.millis(), this)
```

Comment:
ULID instead of UUID v4 would be a better PK for this insert-only schema if possible.